draft-ietf-quic-transport.md

```diff
@@ -1858,6 +1858,12 @@ values for 0-RTT.  This includes initial_max_data and either
 initial_max_streams_bidi and initial_max_stream_data_bidi_remote, or
 initial_max_streams_uni and initial_max_stream_data_uni.
 
+A server might provide larger initial stream flow control limits for streams
+than the remembered values that a client applies when sending 0-RTT.  After
+receiving updated transport parameters, the client updates the flow control
```

Comment:
```suggestion
than the remembered values that a client applies when sending 0-RTT.  Once
the handshake completes, the client updates the flow control
```

I think this is the moment when the client is expected to apply the new transport parameters.

Reply:
My implementation updates when transport parameters are received. Is that wrong?

Could we allow either?

Reply:
In https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#section-7.4.1-2, we state that the transport parameters are updated when the handshake is complete. The requirement for the flow control limits cannot contradict from that.

Also, I tend to believe that applying them at handshake completion is correct, because generally speaking, transport parameters affect application behavior, and application behavior should only depend on messages that have been authenticated.

It might be possible to argue that the new transport parameters are authenticated at the moment they are received because they are AEAD-encrypted using handshake traffic keys that depend on the PSK. But I am not sure if TLS is considered safe when used as such.

Reply:
Good point.  We make them available sooner because if you don't, you can't use 0.5-RTT.  I'll take the suggestion, but maybe we need to take some time to discuss this more at some later point.

Reply:
I was confused by this as well, and had to fix this in quic-go. On the server I now make the transport parameters available immediately in order to support 0.5 RTT, on the client side, I hold them back until handshake completion.

Not really happy with this distinction, but at least it doesn't seem to break anything.